src/OpenTelemetry.Extensions.Hosting/README.md

```diff
@@ -83,6 +83,33 @@ app.Run();
 A fully functional example can be found
 [here](https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/examples/AspNetCore).
 
+### Resources
+
+To dynamically add resources at startup from the dependency injection you can
+provide an `IResourceDetector`.
+To make use of it add it to the dependency injection and they you can use the
+`ISerivceProvider` add it to OpenTelemetry:
+
+```csharp
+public class MyResourceDetector : IResourceDetector
+{
+    public Resource Detect()
+    {
+        return ResourceBuilder.CreateEmpty()
+            .AddService("your service name")
+            .Build();
+    }
+}
```

Comment:
What if we made this slightly more interesting by showing something being injected?

```csharp
public class MyResourceDetector : IResourceDetector
{
    private readonly IWebHostEnvironment webHostEnvironment;

    public MyResourceDetector(IWebHostEnvironment webHostEnvironment)
    {
        this.webHostEnvironment = webHostEnvironment;
    }

    public Resource Detect()
    {
        return ResourceBuilder.CreateEmpty()
            .AddService(serviceName: this.webHostEnvironment.ApplicationName)
            .AddAttributes(new Dictionary<string, object> { ["host.environment"] = this.webHostEnvironment.EnvironmentName })
            .Build();
    }
}
```

Reply:
The environment is injected, I can add something else too if you want.

Reply:
@cromefire I don't see any changes on the diff did you forget to push maybe? 😄

Reply:
No I'm just too stupid to use the GitHub app... Will fix

Reply:
Fixed